### PR TITLE
Uncommitted GC system test

### LINKED ISSUE #4605: Uncommitted GC integration test
Esti test - currently S3
1. Pre - committed and uncommitted data.
2. Run Uncommitted GC
3. Post - verify we keep only committed data.

### PR DESCRIPTION
**Pull request by nopcoder**
Close #4605

### CODE REVIEW

**Review by itaiad200 — APPROVED** [review]
Looks great! We can easily expand the test-cases using this infra. See some non-blocking comments

File: .github/workflows/esti.yaml [hunk 1]
```diff
@@ -177,6 +176,151 @@ jobs:
             echo "Image doesn't exist"
             docker push $ECR_REGISTRY/$ECR_REPOSITORY_LAKEFS:${{ steps.version.outputs.tag }}
           fi
+       
+  login-to-amazon-ecr:
```
[the remark this answers is not in the document]
Comment (nopcoder) [reply .github/workflows/esti.yaml:180]: the service pull the docker image - it require user/password to access our ecr private registry. was hard to find this one.

File: .github/workflows/esti.yaml [hunk 2]
```diff
@@ -177,6 +176,151 @@ jobs:
...
+
+      - name: Run uncommitted gc
+        working-directory: clients/spark
+        run: >
+          CLIENT_JAR=$(ls target/core-301/scala-2.12/lakefs-spark-client-301-assembly-*.jar);
+          echo $CLIENT_JAR; 
+          docker run --network host --add-host lakefs:127.0.0.1 -v $(pwd):/local --rm docker.io/bitnami/spark:3.2.3 spark-submit
+          --master spark://localhost:7077
+          --conf spark.hadoop.lakefs.api.url=http://lakefs:8000/api/v1
+          --conf spark.hadoop.fs.s3.impl=org.apache.hadoop.fs.s3a.S3AFileSystem
```
[the remark this answers is not in the document]
Comment (nopcoder) [reply .github/workflows/esti.yaml:302]: Just to be on the safe side as we access though s3a and s3

File: esti/ugc_test.go [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+		}, "", "  ")
+	if err != nil {
+		t.Fatalf("failed to marshal findings: %s", err)
+	}
+	t.Logf("Findings: %s", findings)
+	findingsPath := findingFilePath()
+	err = os.WriteFile(findingsPath, findings, 0o666)
+	if err != nil {
+		t.Fatalf("Failed to write findings file '%s': %s", findingsPath, err)
+	}
```
[the remark this answers is not in the document]
Comment (nopcoder) [reply esti/ugc_test.go:110]: Nice - but I didn't want to add another underlying file

**Pull request merged**